packages/rule-tester/src/RuleTester.ts

```diff
@@ -0,0 +1,1028 @@
+// Forked from https://github.com/eslint/eslint/blob/ad9dd6a933fd098a0d99c6a9aa059850535c23ee/lib/rule-tester/rule-tester.js
+
+import assert from 'node:assert';
+import path from 'node:path';
+import util from 'node:util';
+
+import type * as ParserType from '@typescript-eslint/parser';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { deepMerge } from '@typescript-eslint/utils/eslint-utils';
+import type {
+  AnyRuleCreateFunction,
+  AnyRuleModule,
+  ParserOptions,
+  RuleContext,
+  RuleModule,
+} from '@typescript-eslint/utils/ts-eslint';
+import { Linter } from '@typescript-eslint/utils/ts-eslint';
+// we intentionally import from eslint here because we need to use the same class
+// that ESLint uses, not our custom override typed version
+import { SourceCode } from 'eslint';
+import merge from 'lodash.merge';
```

Comment:
I thought about that - but ESLint has this exact import for its rule tester.
So I figured that we'd just use the exact same code to avoid a duplicate install